### PR TITLE
AArch64: Add support for AArch64 in JIT-related files

### PR DESCRIPTION
**Pull request by knn-k**
This commit adds support for AArch64 in JIT-related files.

Signed-off-by: knn-k <konno@jp.ibm.com>

### CODE REVIEW

**Comment by knn-k** [comment]
The change in runtime/compiler/module.xml suppresses building the JIT for the time being.

**Comment by knn-k** [comment]
Added changes in jilconsts.c.

File: runtime/oti/JITInterface.hpp [hunk 1]
```diff
@@ -177,6 +186,8 @@ class VM_JITInterface
 			/* Virtual - unsigned offset is in the low 12 bits, assume the sign bit is set (i.e. the offset is always negative) */
 			jitVTableIndex = 0 - (instruction & 0x00000FFF);
 		}
+#elif defined(J9VM_ARCH_AARCH64)
+		/* To be filled */
```
Comment (DanHeidinga) [inline runtime/oti/JITInterface.hpp]: Can these assert / #error for the time being for the same reasons as above?

File: runtime/compiler/module.xml [hunk 2]
```diff
@@ -23,6 +23,7 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 <module>
 	<artifact type="target" name="j9jitlauncher">
 		<include-if condition="spec.flags.interp_nativeSupport"/>
+		<exclude-if condition="spec.linux_aarch64.*" />
```
[the remark this answers is not in the document]
Comment (knn-k) [reply runtime/compiler/module.xml]: Yes, it is only temporary.  I need the line until the JIT for aarch64 can be built.
I am going to add a comment.

File: runtime/compiler/runtime/MethodMetaData.c [hunk 3]
```diff
@@ -1746,6 +1746,8 @@ void jitAddSpilledRegisters(J9StackWalkState * walkState, void * stackMap)
          }
       while (savedGPRs != 0);
       }
+#elif defined(TR_HOST_ARM64)
+   // To be filled
```
[the remark this answers is not in the document]
Comment (knn-k) [reply runtime/compiler/runtime/MethodMetaData.c]: I need it for compiling, and I am going to add assert.

File: runtime/oti/JITInterface.hpp (hunk 1 above)
Comment (knn-k) [reply runtime/oti/JITInterface.hpp]: I am going to add assert here, too.

File: runtime/codert_vm/cnathelp.cpp [hunk 4]
```diff
@@ -247,6 +247,13 @@ J9_EXTERN_BUILDER_SYMBOL(icallVMprJavaSendStatic1);
 #define JIT_LO_U64_RETURN_REGISTER gpr.numbered[0]
 #define JIT_HI_U64_RETURN_REGISTER gpr.numbered[1]
 
+#elif defined(J9VM_ARCH_AARCH64)
+
+#define JIT_STACK_OVERFLOW_SIZE_REGISTER gpr.numbered[18] // To be decided
```
[the remark this answers is not in the document]
Comment (knn-k) [reply runtime/codert_vm/cnathelp.cpp]: Changed the comment.

**Comment by knn-k** [comment]
Updated the files reflecting the review comments.

I used `assert(0)` for adding asserts:
- `MethodMetaData.c` uses `assert()` in other functions in it.
- `JITInterface.hpp` is included by both `codert_vm/cnathelp.cpp` and `vm/BytecodeInterpreter.hpp`.  I don't know which Trace Assert to use in such a case.

**Comment by knn-k** [comment]
#4487 contains some other changes I had to make for building the aarch64 runtime.

**Comment by knn-k** [comment]
I think this can be merged if my changes after the review is OK.

File: runtime/compiler/module.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanHeidinga) [reply runtime/compiler/module.xml]: Can you tag the TODO against an issue that will be closed when it's resolved?  Other wise it's difficult to determine when a TODO in the code has been resolved.

File: runtime/codert_vm/cnathelp.cpp [hunk 5]
```diff
@@ -247,6 +247,13 @@ J9_EXTERN_BUILDER_SYMBOL(icallVMprJavaSendStatic1);
 #define JIT_LO_U64_RETURN_REGISTER gpr.numbered[0]
 #define JIT_HI_U64_RETURN_REGISTER gpr.numbered[1]
 
+#elif defined(J9VM_ARCH_AARCH64)
+
+#define JIT_STACK_OVERFLOW_SIZE_REGISTER gpr.numbered[18] // TODO: determine which register to use
```
Comment (DanHeidinga) [inline runtime/codert_vm/cnathelp.cpp]: Can you tie the //TODO to an issue #?

File: runtime/compiler/module.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (knn-k) [reply runtime/compiler/module.xml]: I opened Issue #4854 for this.

File: runtime/codert_vm/cnathelp.cpp (hunk 5 above)
Comment (knn-k) [reply runtime/codert_vm/cnathelp.cpp]: I opened Issue #4855 for this.

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk8

**Comment by knn-k** [comment]
Fixed runtime/ddr/jitflagsddr.c.

**Comment by DanHeidinga** [comment]
Build failed due to:
```
10:28:00  In file included from jitflagsddr.c:24:0:
10:28:00  jitflagsddr.c:44:50: error: ‘host_AArch64’ undeclared here (not in a function); did you mean ‘host_AARCH64’?
10:28:00   J9DDRConstantTableEntryWithValue("host_AArch64", host_AArch64)
```

**Comment by knn-k** [comment]
I have already fixed the problem with `jitflagsddr.c` in Jenkins build.

The Travis CI build is failing with the following error:
> * For target jdk_modules_java.base__the.java.base_batch:
> Connection attempt failed: Connection refused (Connection refused)
> Connection attempt failed: Connection refused (Connection refused)
> Connection attempt failed: Connection refused (Connection refused)
> Giving up
> IOException caught during compilation: Could not connect to server

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk8

**Pull request merged**